### PR TITLE
Account info

### PR DESCRIPTION
**Pull request by dhodcz2**
Account class with data encapsulation of WalletAddress, Balance, and StateNonce. There is data validation for WalletAddress that raises an exception if an illegal string is passed.

### CODE REVIEW

File: client/src/main/java/sigblockchain/projectred/client/Account.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package sigblockchain.projectred.client;
+
+import java.math.BigInteger;
+
+public class Account {
+	private String WalletAddress;
```
Comment (Vineet77) [inline client/src/main/java/sigblockchain/projectred/client/Account.java:6]: By convention, Java instance variables start with lower case characters. 
https://www.oracle.com/technetwork/java/codeconventions-135099.html

File: client/src/main/java/sigblockchain/projectred/client/Account.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	private BigInteger Balance;
+	private BigInteger StateNonce;
+
+	public Account(String walletAddress, BigInteger balance, BigInteger stateNonce) {
+		validateWalletAddress(walletAddress);
+
+		WalletAddress = walletAddress;
+		Balance = balance;
```
Comment (Vineet77) [inline client/src/main/java/sigblockchain/projectred/client/Account.java:14]: When assigning instance variables, it should be of the form `this.variable = value`.

File: client/src/main/java/sigblockchain/projectred/client/Account.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import java.math.BigInteger;
+
+public class Account {
+	private String WalletAddress;
+	private BigInteger Balance;
+	private BigInteger StateNonce;
+
+	public Account(String walletAddress, BigInteger balance, BigInteger stateNonce) {
+		validateWalletAddress(walletAddress);
```
Comment (Vineet77) [inline client/src/main/java/sigblockchain/projectred/client/Account.java:11]: For now you don't need to validate the wallet address. But you do need to validate balance and state nonce, as neither value can be negative.

File: client/src/main/java/sigblockchain/projectred/client/Account.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+	public Account(String walletAddress) {
+		validateWalletAddress(walletAddress);
+
+		WalletAddress = walletAddress;
+		Balance = BigInteger.ZERO;
+		StateNonce = BigInteger.ZERO;
+	}
+
+	private void validateWalletAddress(String walletAddress) {
```
Comment (Vineet77) [inline client/src/main/java/sigblockchain/projectred/client/Account.java:26]: This seems to be the standard for bitcoin. Aurum does not follow those standards.

File: client/src/main/java/sigblockchain/projectred/client/Account.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	public BigInteger getBalance() {
+		return Balance;
+	}
+
+	public BigInteger getStateNonce() {
+		return StateNonce;
+	}
+
+	public void setBalance(BigInteger balance) {
+		Balance = balance;
```
Comment (Vineet77) [inline client/src/main/java/sigblockchain/projectred/client/Account.java:56]: You need to check if the balance isn't being set to a negative value.

**Review by Vineet77 — CHANGES_REQUESTED** [review]
Please implement the changes suggested. Also the pull request should be `dev <- account_info`. Currently this pull request is `master <- acccount_info`. I'm closing this pull request. Once the changes are implemented and tests pass, create another pull request to the `dev` branch.

File: client/src/main/java/sigblockchain/projectred/client/Account.java (hunk 3 above)
Comment (dhodczak) [reply client/src/main/java/sigblockchain/projectred/client/Account.java:11]: Would you rather I comment out the wallet address validation as a placeholder for the future, or remove it altogether?

**Pull request closed without merging**